gulp/tasks/compare.js

```diff
@@ -26,8 +28,9 @@ gulp.task('compare', function (done) {
 
             // if the test report is enabled in the config
             if (testSuite) {
-                junitWriter.save(path.join(paths.ci_report, 'xunit.xml'), function() {
-                    console.log('\x1b[32m', 'Regression test report file (xunit.xml) is successfully created.', '\x1b[0m');
+                testReportFileName = paths.ciReport.testReportFileName.replace(/(\..+)?$/, '.xml');
```

Comment:
I'm inclined to just let the user name it whatever they want _or_ if you really want to enforce/signal file type, just _append_ an `.xml` extension if one doesn't exist.  (As opposed to clobbering the user's input)